lib/Sema/CSRanking.cpp

```diff
@@ -1381,3 +1353,61 @@ SolutionDiff::SolutionDiff(ArrayRef<Solution> solutions) {
     }
   }
 }
+
+InputMatcher::InputMatcher(const ValueDecl *funcDecl,
+    const ArrayRef<AnyFunctionType::Param> params) :
+    NumSkippedParameters(0), Params(params) {
+  DefaultValueMap = computeDefaultMap(params, funcDecl,
+      funcDecl->getDeclContext()->isTypeContext());
+}
+
+InputMatcher::Result InputMatcher::solve(int numInputs,
```

Comment:
Maybe it makes sense to rename this to `match` instead of `solve`?

Reply:
Yep!